### PR TITLE
[179] add download component and inlcude in teaching_assistant resource

### PR DESCRIPTION
**Pull request by RobertM905**

### CODE REVIEW

File: lib/gateway/contentful_gateway.rb [hunk 1]
```diff
@@ -153,12 +153,28 @@ def parse_content(content)
...
+    when 'download'
+      create_download(content)
     else
       @logger.warn("Content #{content.sys[:content_type].id} not supported")
       nil
     end
   end
 
+  def create_download(content)
+    puts content.download_resource.file.url
```
Comment (chubberlisk) [inline lib/gateway/contentful_gateway.rb]: A lil boo boo. 👀

File: lib/gateway/contentful_gateway.rb [hunk 2]
```diff
@@ -153,12 +153,28 @@ def parse_content(content)
...
+    {
+      type: :download,
+      data: {
+        name: content.name,
+        thumbnail_image_src: content.thumbnail_image.url,
+        download_resource_link: content.download_resource.url,
```
Comment (chubberlisk) [inline lib/gateway/contentful_gateway.rb]:
```suggestion
        resource_link: content.resource.url,
```

I think `download` is unnecessary as it's a type field for a `download` type. Same thoughts for `download_resource_type` and `download_link_text`.

File: lib/gateway/contentful_gateway.rb [hunk 3]
```diff
@@ -153,12 +153,28 @@ def parse_content(content)
...
+        download_resource_type: content.download_resource.file.content_type,
+        download_link_text: content.download_link_text
```
Comment (chubberlisk) [inline lib/gateway/contentful_gateway.rb]: What's the difference between `name` and `download_link_text`?

**Review by chubberlisk — COMMENTED** [review]
Mostly looks good to me, just some naming suggestions and a question.

File: lib/gateway/contentful_gateway.rb (hunk 1 above)
Comment (RobertM905) [reply lib/gateway/contentful_gateway.rb]: Nice catch thanks!

File: lib/gateway/contentful_gateway.rb (hunk 2 above)
Comment (RobertM905) [reply lib/gateway/contentful_gateway.rb]: Makes sense. Ive changed it in the gateway, component renderer and on contentful 👍

File: lib/gateway/contentful_gateway.rb (hunk 3 above)
Comment (RobertM905) [reply lib/gateway/contentful_gateway.rb]: content.name was more important for the contentful side and is not used on the component side. I have removed it from the gateway.

**Review by chubberlisk — APPROVED** [review]
Looks good to me!

**Pull request merged**